OpenRA.Game/Actor.cs

```diff
@@ -47,8 +47,9 @@ internal struct SyncHash
 
 		public int Generation;
 
-		public Rectangle Bounds { get; private set; }
-		public Rectangle VisualBounds { get; private set; }
+		public Rectangle RenderBounds { get; private set; }
+		public Rectangle SelectableBounds { get; private set; }
+		public Rectangle SelectionOverlayBounds { get; private set; }
```

Comment:
Can you please move this to the `Selectable` trait?  The other two are needed by engine level code, but this one is purely visual.

One of the things on my Next+1 todo list is to rework the Selectable trait to support mod-defined selection boxes and non-rectangular click regions.  The changes here otherwise nicely prepare for this.

Reply:
> Can you please move this to the Selectable trait? The other two are needed by engine level code, but this one is purely visual.

This requires quite a bunch of changes and `Selectable` trait look-ups, of which at least the one in `SelectionBarsRenderable` can't be cached, as far as I can tell. Are you sure?

Reply:
You should be able to pass the trait to the constructor alongside the actor.

Reply:
Leaving as-is for now as discussed in IRC.